hack/generator/pkg/astmodel/object_type.go

```diff
@@ -63,6 +66,75 @@ func (objectType *ObjectType) generateMethodDecls(codeGenerationContext *CodeGen
 	return result
 }
 
+func (objectType *ObjectType) generateInterfaceDecls(codeGenerationContext *CodeGenerationContext, typeName TypeName) []ast.Decl {
+	var result []ast.Decl
+
+	// First interfaces must be ordered by name for deterministic output
+	var ifaceNames []TypeName
+	for ifaceName := range objectType.interfaces {
+		ifaceNames = append(ifaceNames, ifaceName)
+	}
+
+	sort.Slice(ifaceNames, func(i int, j int) bool {
+		return ifaceNames[i].name < ifaceNames[j].name
+	})
+
+	for _, ifaceName := range ifaceNames {
+		iface := objectType.interfaces[ifaceName]
+
+		result = append(result, objectType.generateInterfaceImplAssertion(codeGenerationContext, iface, typeName))
+
+		var funcNames []string
+		for funcName := range iface.functions {
+			funcNames = append(funcNames, funcName)
+		}
+
+		sort.Slice(funcNames, func(i int, j int) bool {
+			return funcNames[i] < funcNames[j]
+		})
+
+		for _, methodName := range funcNames {
+			function := iface.functions[methodName]
+			result = append(result, function.AsFunc(codeGenerationContext, typeName, methodName))
+		}
+	}
+
+	return result
+}
+
+func (objectType *ObjectType) generateInterfaceImplAssertion(codeGenerationContext *CodeGenerationContext, iface *Interface, typeName TypeName) ast.Decl {
```

Comment:
Great to have this.

Reply:
Yes, I like the expression of intent in the code, even in generated code where we can "guarantee" a struct impls an interface since we control the generation process.